backend_new/src/utilities/build-order-by.ts

```diff
@@ -1,4 +1,49 @@
+import { ListingOrderByKeys } from '../enums/listings/order-by-enum';
 import { OrderByEnum } from '../enums/shared/order-by-enum';
+import { Prisma } from '@prisma/client';
+
+/* 
+  Constructs the "orderBy" part of the prisma query and maps the values to
+  the appropriate listing field
+*/
+export const buildOrderByForListings = (
+  orderBy?: string[],
+  orderDir?: OrderByEnum[],
+): Prisma.ListingsOrderByWithRelationInput[] => {
+  if (!orderBy?.length) {
+    return undefined;
+  }
+
+  return orderBy.map((param, index) => {
+    switch (param) {
+      case ListingOrderByKeys.mostRecentlyUpdated:
+        return { updatedAt: orderDir[index] };
+      case ListingOrderByKeys.status:
+        return { status: orderDir[index] };
+      case ListingOrderByKeys.name:
+        return { name: orderDir[index] };
+      case ListingOrderByKeys.waitlistOpen:
+        return { isWaitlistOpen: orderDir[index] };
+      case ListingOrderByKeys.unitsAvailable:
+        return { unitsAvailable: orderDir[index] };
+      case ListingOrderByKeys.mostRecentlyClosed:
+        return {
+          closedAt: orderDir[index],
+        };
+      case ListingOrderByKeys.mostRecentlyPublished:
+        return {
+          publishedAt: orderDir[index],
+        };
+      case ListingOrderByKeys.marketingType:
+        return { marketingType: orderDir[index] };
+      case ListingOrderByKeys.applicationDates:
+      case undefined:
+        // Default to ordering by applicationDates (i.e. applicationDueDate
+        // and applicationOpenDate) if no orderBy param is specified.
+        return { applicationDueDate: orderDir[index] };
```

Comment:
is there ever a case for orderDir missing for this iteration?

Reply:
I added an additional check to make sure that the orderDir is the same length as the type